In [2]:
import pandas as pd

__Вопрос:__ Откройте файл с таблицей Выведите последние 10 строк.
Посмотрите на данные и скажите, что они из себя представляют, сколько в таблице строк, какие столбцы?

In [18]:
data = pd.read_csv('data.csv')
display(data.tail(10))
display(data.info())

order_id  quantity            item_name  \
4612      1831         1        Carnitas Bowl   
4613      1831         1                Chips   
4614      1831         1        Bottled Water   
4615      1832         1   Chicken Soft Tacos   
4616      1832         1  Chips and Guacamole   
4617      1833         1        Steak Burrito   
4618      1833         1        Steak Burrito   
4619      1834         1   Chicken Salad Bowl   
4620      1834         1   Chicken Salad Bowl   
4621      1834         1   Chicken Salad Bowl   

                                     choice_description item_price  
4612  [Fresh Tomato Salsa, [Fajita Vegetables, Rice,...     $9.25   
4613                                                NaN     $2.15   
4614                                                NaN     $1.50   
4615   [Fresh Tomato Salsa, [Rice, Cheese, Sour Cream]]     $8.75   
4616                                                NaN     $4.45   
4617  [Fresh Tomato Salsa, [Rice, Black Beans, Sour ...    $11.75   
4618  [Fresh Tomato Salsa, [Rice, Sour Cream, Cheese...    $11.75   
4619  [Fresh Tomato Salsa, [Fajita Vegetables, Pinto...    $11.25   
4620  [Fresh Tomato Salsa, [Fajita Vegetables, Lettu...     $8.75   
4621  [Fresh Tomato Salsa, [Fajita Vegetables, Pinto...     $8.75

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 4622 entries, 0 to 4621
Data columns (total 5 columns):
 #   Column              Non-Null Count  Dtype 
---  ------              --------------  ----- 
 0   order_id            4622 non-null   int64 
 1   quantity            4622 non-null   int64 
 2   item_name           4622 non-null   object
 3   choice_description  3376 non-null   object
 4   item_price          4622 non-null   object
dtypes: int64(2), object(3)
memory usage: 180.7+ KB


None

__Ответ:__ В таблице 4622 строки и 5 столбцов. Столбцы показывают соответственно номер заказа, количество конкретного блюда в заказе, название этого самого блюда, выбранные ингредиенты (если доступно), цена блюда. Просмотрев содержимое, можем заключить, что в этой таблице представлены по заказам в неком ресторане мексиканской еды.

__Вопрос:__ Сколько заказов попало в выборку?
Сколько уникальных категорий товара было куплено? (item_name)

In [43]:
print(f"Заказов в выборке: {len(data['order_id'].unique())}")
print(f"Уникальных категорий товара: {len(data['item_name'].unique())}")

Заказов в выборке: 1834
Уникальных категорий товара: 50


__Вопрос:__ Есть ли в данных пропуски? В каких колонках?   
Заполните пропуски пустой строкой для строковых колонок и нулём для числовых.

__Ответ:__ Как видно из вызова метода `info`, пропуски есть только в колонке 'choice_description'. Но на всякий случай напишем решение для всех колонок. 

In [46]:
for column in data:
    data[column] = (data[column].fillna(0) if data[column].dtype == 'int64' else data[column].fillna(''))
data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 4622 entries, 0 to 4621
Data columns (total 5 columns):
 #   Column              Non-Null Count  Dtype 
---  ------              --------------  ----- 
 0   order_id            4622 non-null   int64 
 1   quantity            4622 non-null   int64 
 2   item_name           4622 non-null   object
 3   choice_description  4622 non-null   object
 4   item_price          4622 non-null   object
dtypes: int64(2), object(3)
memory usage: 180.7+ KB


__Вопрос:__ Какая средняя/минимальная/максимальная цена у товара?

Разбейте заказы на три группы: дешевые, средние и дорогие. Порог для групп подберите сами (посмотрите, на около какой суммы чаще берут в каждой группе).